0001-typedb/02-dev/03-match.md

```diff
@@ -0,0 +1,792 @@
+| relation                            | `[$<var_label>] ([<role>:] $<var_label> [, [<role>:] $<var_label>]...) isa <relation-type> [, <has-attribute expression>...];`  |
+| has-attribute expression            | `has <attr-type> ($<var_label> &#124; [<comparison-operator>] <value> );`                                                       |
+| comparison                          | `$<var_label> [<comparison-operator>] <value>;`                                                                                 |
+| equal                               | `$<var_label> is $<var_label>;`                                                                                                 |
+
+### Comparison operators
+
+The following operators are supported for comparing attribute values: `=`, `!=`, `>`, `>=`, `<` and `<=`.
+
+### Combining statements
+
+![Combining statements](../../images/query/pattern-structure.png)
+
+By arranging statements together, we can express more complex pattern scenarios and their corresponding data.
+
+- Statement: simplest possible arrangement — a single basic building block as [explained above](#patterns-overview).
+- Conjunction (logical `AND`): a set of statements where to satisfy a match, all statements must be true.
+  We use conjunctions by default just by separating the partaking statements with semicolons `;`.
+- Disjunction (logical `OR`): a set of statements where to satisfy a match, at least one statement must be matched. 
+  We form disjunctions by enclosing the partaking statements within curly braces `{}` and joining them together with the 
+  keyword `or`.
+- Negation (logical negation): a statement that explicitly defines conditions that must **not** to be met. We form 
+  negations by defining the conditions not to be met in curly brackets of a `not {};` block.
+
+## Match clause
+
+### Schema queries
+
+A `match` clause can be used to find types or roles in a database schema.
+
+#### Type matching
+
+##### All types
+
+Use a `subtype` pattern with type `thing` to find all types defined in a schema.
+
+<!-- test-ignore -->
+```typeql
+match $t sub thing;
+```
+
+<div class="note">
+[Warning]
+The `thing` base type will be deprecated in TypeDB version 3.0. Consider using `entity`, `attribute`, or `relation` 
+base type instead. To produce the same result as the above example, use the following query:
+
+<!-- test-ignore -->
+```typeql
+match $s sub $t; { $t type entity; } or { $t type relation; } or { $t type attribute; };
+```
+
+</div>
+
+##### Specific type or nested subtype
+
+Use a `subtype` pattern to find a specific type and all of its subtypes.
+
+<!-- test-ignore -->
+```typeql
+match $o sub object;
+```
+
+The above query finds the `object` type and all of its nested subtypes: direct (i.e. `resource` and `resource-collection`) 
+and indirect (i.e. `file`, `interface`, `directory` and `application`).
```

Comment:
It's not that important, since we already declared how queries built with match clause, but yes, makes sense to try avoid going to query territory yet.
```suggestion
The above pattern finds the `object` type and all of its nested subtypes: direct (i.e. `resource` and `resource-collection`) 
and indirect (i.e. `file`, `interface`, `directory` and `application`).
```